### PR TITLE
Merge with FF 46.0

### PR DESCRIPTION
**Pull request by alver-cliqz**
Please, check second commit here - it contain all conflict resolution and some fixes to make browser buildable again.

@luciancor, @chrmod, @maxim-cliqz 
General info:
1. FF continue removing pre-processing for jsm, jx, xul, etc. files. We need to be careful with #if directive
2. A lot of changes in their tracking protection system. We need to re-test DB-527.

### CODE REVIEW

**Comment by luciancor** [comment]
LGTM

File: mozilla-release/browser/components/nsBrowserGlue.js [hunk 1]
```diff
@@ -1222,47 +1192,17 @@ BrowserGlue.prototype = {
...
       let skipDefaultBrowserCheck = false;
       if (!AppConstants.RELEASE_BUILD) {
         promptCount =
           Services.prefs.getIntPref("browser.shell.defaultBrowserCheckCount");
-        skipDefaultBrowserCheck =
-          Services.prefs.getBoolPref("browser.shell.skipDefaultBrowserCheck");
       }
+      // CLIQZ: will skip showing dialog "Set as default" on first launch
+      skipDefaultBrowserCheck =
+        Services.prefs.getBoolPref("browser.shell.skipDefaultBrowserCheck");
```
Comment (maxim-cliqz) [inline mozilla-release/browser/components/nsBrowserGlue.js]: I'd leave in inside the original block, inverse the condition, and `#if 0` the `promptCount` variable. That would minimize the difference to Mozilla's code.

File: mozilla-release/browser/components/preferences/in-content/privacy.xul [hunk 2]
```diff
@@ -170,16 +138,12 @@
               preference="pref.privacy.disable_button.change_blocklist"/>
     </hbox>
   </vbox>
-<<<<<<< HEAD
-#endif
-||||||| merged common ancestors
-=======
   <vbox>
```
Comment (maxim-cliqz) [inline mozilla-release/browser/components/preferences/in-content/privacy.xul]: It seems, we don't want to hide this block (it was refactored from `checkbox id="privacyDoNotTrackCheckbox"` above and moved down here). It's related to DoNotTrack header preference, which we still keep.

File: mozilla-release/browser/locales/en-US/chrome/browser/browser.dtd [hunk 3]
```diff
@@ -704,14 +692,14 @@ you can use these alternative items. Otherwise, their values should be empty.  -
...
+<!ENTITY quitApplicationCmd.label       "Quit">
 <!ENTITY quitApplicationCmd.accesskey   "Q">
 <!ENTITY quitApplicationCmdMac2.label   "Quit &brandShorterName;">
 <!-- LOCALIZATION NOTE(quitApplicationCmdUnix.key): This keyboard shortcut is used by both Linux and OSX builds. -->
 <!ENTITY quitApplicationCmdUnix.key     "Q">
 
-<!ENTITY closeCmd.label                 "Close">  
-<!ENTITY closeCmd.key                   "W">  
+<!ENTITY closeCmd.label                 "Close">
+<!ENTITY closeCmd.key                   "W">
```
Comment (maxim-cliqz) [inline mozilla-release/browser/locales/en-US/chrome/browser/browser.dtd]: Lots of parasitic diffs and lost authorship :(

File: mozilla-release/browser/locales/en-US/chrome/browser/preferences/privacy.dtd [hunk 4]
```diff
@@ -13,24 +13,7 @@
 <!ENTITY  trackingProtectionExceptions.label   "Exceptions…">
 <!ENTITY  trackingProtectionExceptions.accesskey "x">
 
-<<<<<<< HEAD
-<!ENTITY dntTrackingNotOkay4.label     "Request that sites not track you">
-<!ENTITY dntTrackingNotOkay4.accesskey "n">
-<!ENTITY doNotTrackInfo.label          "Learn More">
-<!ENTITY doNotTrackInfo.label2         "When the feature is enabled, CLIQZ will tell advertising networks and other websites and applications that you want to opt-out of tracking for purposes like behavioral advertising.">
```
Comment (maxim-cliqz) [inline mozilla-release/browser/locales/en-US/chrome/browser/preferences/privacy.dtd]: We should check, what's instead of this line now.

**Comment by maxim-cliqz** [comment]
LGTM

**Pull request merged**